Assets/Scripts/RestMovesOpponent.cs

```diff
@@ -1,156 +1,124 @@
+                        continue;
+                    }
+
+                    GoToField(numbersField, gameCalculator, zeroes, moveCounter);
+                    return;
+                }
+            }
         }
-        else if (CheckCornerField(new[] {3, 0, 7, 8, 4, 2, 6}, FieldValue.Zero))
-        {
-            numberField = 6;
+    }
 
-            GoToField(numberField);
-        }
-        else if (CheckCornerField(new[] {6, 7, 5, 2, 4, 0, 8}, FieldValue.Zero))
-        {
-            numberField = 8;
+    private void GoToField(int numberField, GameCalculator gameCalculator, List<GameObject> zeroes,
+        MoveCounter moveCounter)
+    {
+        gameCalculator._fieldValues[numberField] = FieldValue.Zero;
 
-            GoToField(numberField);
-        }
-        else if (CheckIntermediateField(new[] {4, 7, 0, 2, 1}, FieldValue.Zero))
-        {
-            numberField = 1;
+        zeroes[numberField].SetActive(true);
 
-            GoToField(numberField);
-        }
-        else if (CheckIntermediateField(new[] {3, 4, 8, 2, 5}, FieldValue.Zero))
-        {
-            numberField = 5;
+        moveCounter.IncreaseCount();
+    }
 
-            GoToField(numberField);
-        }
-        else if (CheckIntermediateField(new[] {4, 5, 6, 0, 3}, FieldValue.Zero))
-        {
-            numberField = 3;
+    private bool CheckIntermediateField(int[] numbers, FieldValue fieldValue, GameCalculator gameCalculator)
+    {
+        return (gameCalculator._fieldValues[numbers[0]] == fieldValue &&
+                gameCalculator._fieldValues[numbers[1]] == fieldValue ||
+                gameCalculator._fieldValues[numbers[2]] == fieldValue &&
+                gameCalculator._fieldValues[numbers[3]] == fieldValue) &&
+               gameCalculator._fieldValues[numbers[4]] == FieldValue.Empty;
+    }
 
-            GoToField(numberField);
-        }
-        else if (CheckIntermediateField(new[] {1, 4, 6, 8, 7}, FieldValue.Zero))
-        {
-            numberField = 7;
+    private bool CheckCornerField(int[] numbers, FieldValue fieldValue, GameCalculator gameCalculator)
+    {
+        return (gameCalculator._fieldValues[numbers[0]] == fieldValue &&
+                gameCalculator._fieldValues[numbers[1]] == fieldValue ||
+                gameCalculator._fieldValues[numbers[2]] == fieldValue &&
+                gameCalculator._fieldValues[numbers[3]] == fieldValue ||
+                gameCalculator._fieldValues[numbers[4]] == fieldValue &&
+                gameCalculator._fieldValues[numbers[5]] == fieldValue) &&
+               gameCalculator._fieldValues[numbers[6]] == FieldValue.Empty;
+    }
 
-            GoToField(numberField);
-        }
 
-        else if (CheckCornerField(new[] {0, 1, 8, 5, 6, 4, 2}, FieldValue.Cross))
+    private bool CheckFieldsWithValues(FieldValue fieldValue, GameCalculator gameCalculator, List<GameObject> zeroes,
+        MoveCounter moveCounter)
+    {
+        int numberField = VerificationNumber;
```

Comment:
wtf is verification number

Reply:
If you are using it as flag that something happened you should revise your logic or just use boolean value for it.